app/components/channel_intro/channel_intro.js

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PropTypes, PureComponent} from 'react';
+import {
+    StyleSheet,
+    Text,
+    View
+} from 'react-native';
+import {getFullName} from 'mattermost-redux/utils/user_utils';
+import {Constants} from 'mattermost-redux/constants';
+import {injectIntl, intlShape} from 'react-intl';
+
+import ProfilePicture from 'app/components/profile_picture';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+class ChannelIntro extends PureComponent {
+    static propTypes = {
+        currentChannel: PropTypes.object.isRequired,
+        currentChannelMembers: PropTypes.array.isRequired,
+        currentUser: PropTypes.object.isRequired,
+        intl: intlShape.isRequired,
+        theme: PropTypes.object.isRequired
+    };
+
+    getDisplayName = (member) => {
+        if (!member) {
+            return null;
+        }
+
+        const displayName = getFullName(member);
+
+        if (!displayName) {
+            return member.username;
+        }
+
+        return displayName;
+    };
+
+    buildProfiles = () => {
+        const {currentChannelMembers, theme} = this.props;
+        const style = getStyleSheet(theme);
+
+        return currentChannelMembers.map((member) => (
+            <View
+                key={member.id}
+                style={style.profile}
+            >
+                <ProfilePicture
+                    user={member}
+                    size={64}
+                    statusBorderWidth={2}
+                    statusSize={25}
+                    statusIconSize={15}
+                />
+            </View>
+        ));
+    }
+
+    buildNames = () => {
+        const {currentChannelMembers, theme} = this.props;
+        const style = getStyleSheet(theme);
+
+        const names = currentChannelMembers.map((member) => this.getDisplayName(member));
+
+        return <Text style={style.displayName}>{names.join(', ')}</Text>;
+    }
+
+    buildDMContent = () => {
+        const {currentChannelMembers, intl, theme} = this.props;
+        const style = getStyleSheet(theme);
+        const teammate = this.getDisplayName(currentChannelMembers[0]);
+
+        return (
+            <Text style={style.message}>
+                {intl.formatMessage({
+                    id: 'mobile.intro_messages.DM',
+                    defaultMessage: 'This is the start of your direct message history with {teammate}. Direct messages and files shared here are not shown to people outside this area.'
+                }, {
+                    teammate
+                })}
+            </Text>
+        );
+    }
+
+    buildGMContent = () => {
+        const {intl, theme} = this.props;
+        const style = getStyleSheet(theme);
+
+        return (
+            <Text style={style.message}>
+                {intl.formatMessage({
+                    id: 'intro_messages.group_message',
```

Comment:
@jarredwitt I think we are missing another option here that in platform is `intro_messages.GM`